libs/auth-nest-tools/src/lib/auth.middleware.ts

```diff
@@ -18,22 +19,111 @@ interface Middleware {
   pre?(context: RequestContext): Promise<FetchParams | void>
 }
 
+export interface AuthMiddlewareOptions {
+  forwardUserInfo: boolean
+  tokenExchangeOptions?: TokenExchangeOptions
+}
+
+export interface TokenExchangeOptions {
+  issuer: string
+  clientId: string
+  clientSecret: string
+  scope: string
+}
```

Comment:
Would be nice if the token exchange could be short-circuited if the authentication already contains the necessary scopes.

Reply:
Implemented